### ISSUE TITLE
LocalCluster() fails to start Diag Server

### ISSUE THREAD

**Issue by m-ric** [dask/distributed#477]
A simple:
```
import distributed
c = LocalCluster()
```
fails to start diagserver with following error. Running `LocalCluster(diagnostics_port=None)` does not show the error.
```
In [1]: import distributed

In [2]: c = distributed.LocalCluster()
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-2-9953140d5917> in <module>()
----> 1 c = distributed.LocalCluster()

/home/johndoe/.local/lib/python2.7/site-packages/distributed/deploy/local.pyc in __init__(self, n_workers, threads_per_worker, nanny, loop, start, scheduler_port, silence_logs, diagnostics_port, services, **kwargs)
     99         if diagnostics_port is not None:
    100             self.start_diagnostics_server(diagnostics_port,
--> 101                                           silence=silence_logs)
    102 
    103     def __str__(self):

/home/johndoe/.local/lib/python2.7/site-packages/distributed/deploy/local.pyc in start_diagnostics_server(self, port, show, silence)
    179                 http_port=self.scheduler.services['http'].port,
    180                 bokeh_port=port, show=show,
--> 181                 log_level=logging.getLevelName(silence).lower())
    182 
    183     @gen.coroutine

/home/johndoe/.local/lib/python2.7/site-packages/distributed/bokeh/application.pyc in __init__(self, host, http_port, tcp_port, bokeh_port, bokeh_whitelist, log_level, show, prefix, use_xheaders, quiet)
     81             import subprocess
     82             self.process = subprocess.Popen(args, stdout=subprocess.PIPE,
---> 83                     stderr=subprocess.PIPE)
     84 
     85         if not quiet:

/usr/lib/python2.7/subprocess.pyc in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags)
    708                                 p2cread, p2cwrite,
    709                                 c2pread, c2pwrite,
--> 710                                 errread, errwrite)
    711         except Exception:
    712             # Preserve original exception in case os.close raises.

/usr/lib/python2.7/subprocess.pyc in _execute_child(self, args, executable, preexec_fn, close_fds, cwd, env, universal_newlines, startupinfo, creationflags, shell, to_close, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite)
   1325                         raise
   1326                 child_exception = pickle.loads(data)
-> 1327                 raise child_exception
   1328 
   1329 

OSError: [Errno 2] No such file or directory
```

On one of my machine running python 2.7.6 on Ubuntu-14.04.4 (x86_64), with following python packages installed locally (--user):
```
$ pip show -V dask pandas distributed numpy cloudpickle bokeh tornado boto3 click msgpack-python locket psutil toolz six s3fs tblib backports.ssl_match_hostname singledispatch certifi backports_abc | grep "Name\|Version"
Name: dask
Version: 0.11.0
Name: pandas
Version: 0.18.0
Name: distributed
Version: 1.12.1
Name: numpy
Version: 1.11.1
Name: cloudpickle
Version: 0.2.1
Name: bokeh
Version: 0.12.1
Name: tornado
Version: 4.4.1
Name: boto3
Version: 1.4.0
Name: click
Version: 6.6
Name: msgpack-python
Version: 0.4.8
Name: locket
Version: 0.2.0
Name: psutil
Version: 4.3.0
Name: toolz
Version: 0.8.0
Name: six
Version: 1.10.0
Name: s3fs
Version: 0.0.7
Name: tblib
Version: 1.3.0
Name: singledispatch
Version: 3.4.0.3
Name: certifi
Version: 2016.8.31
```

**Comment by m-ric** [comment]
It works well on another machine running the same OS and python env. The only difference lies in certifi:
```
Name: certifi
Version: 2015.9.6.2
```
`pip install -I certifi==2015.9.6.2 --user` installed 2016.8.31 for some reason on my faulty machine, but did not heal.

**Comment by mrocklin** [comment]
Hrm, odd.  Regardless we should be robust to a failure of the bokeh/diagnostic server.  It's non-essential.

**Comment by mrocklin** [comment]
https://github.com/dask/distributed/pull/481

**Comment by mrocklin** [comment]
Does the Bokeh server run for you generally if you try one of their examples?

**Comment by mrocklin** [comment]
@m-ric does #488 resolve this issue for you?

**Comment by mrocklin** [comment]
Closing for now.  Please reopen if the issue persists

**Comment by m-ric** [comment]
My problem occurs in `ipython` only. Default `python2.7.6` interpreter is not impacted, i.e. I can access bokeh server properly when `LocalCluster()` is started from default interpreter. I tried distributed master branch containing 9d0642d commit, to no avail. I believe it comes from an issue with my python environment on this specific machine. Note that I installed nearly all my python modules through `pip instlal <pkg> --user`.
